src/cli/main.c

```diff
@@ -88,35 +88,69 @@ parse_profile_options(struct cli_cmdline *cmdline,
 
 static errno_t activate(struct cli_cmdline *cmdline)
 {
+    struct authselect_profile *profile = NULL;
+    const char **features = NULL;
     const char *profile_id;
-    const char **features;
+    const char *requirements;
     int enforce = 0;
+    int quiet = 0;
     errno_t ret;
 
     struct poptOption options[] = {
         {"force", 'f', POPT_ARG_VAL, &enforce, 1, _("Enforce changes"), NULL },
+        {"quiet", 'q', POPT_ARG_VAL, &quiet, 1, _("Do not print profile requirements"), NULL },
         POPT_TABLEEND
     };
 
     ret = parse_profile_options(cmdline, options, &profile_id, &features);
     if (ret != EOK) {
-        return ret;
+        goto done;
+    }
+
+    ret = authselect_profile(profile_id, &profile);
+    if (ret != EOK) {
+        ERROR("Unable to get profile information [%d]: %s",
+              ret, strerror(ret));
+        ret = ENOMEM;
+        goto done;
+    }
+
+    requirements = authselect_profile_requirements(profile, features);
+    if (requirements == NULL) {
+        ERROR("Unable to read profile requirements!");
+        ret = EFAULT;
+        goto done;
     }
 
     ret = authselect_activate(profile_id, features, enforce);
-    free(features);
     if (ret == EEXIST) {
         fprintf(stderr, _("\nSome unexpected changes to the configuration were "
                 "detected.\nUse --force parameter if you want to overwrite "
                 "these changes.\n"));
-        return ret;
+        goto done;
     } else if (ret != EOK) {
         fprintf(stderr, _("Unable to activate profile [%d]: %s\n"),
                 ret, strerror(ret));
-        return ret;
+        goto done;
     }
 
-    return EOK;
+    if (!quiet) {
+        printf(_("Profile \"%s\" was selected.\n"), profile_id);
+
+        if (requirements[0] != '\0') {
+            printf("\n%s\n", requirements);
+        }
+    }
+
+    ret = EOK;
+
+done:
+    authselect_profile_free(profile);
+    if (features != NULL) {
+        free(features);
```

Comment:
freeing a NULL pointer is legal in C, so the if is not needed

Reply:
Thought there were some issues in some systems, but from what I read now it is part of standard and we are safe on Linux anyway. Thanks.

Reply:
I will leave it on other places as it is already there though.

Reply:
On the other hands, I decided to leave it there for now so the code is consistent and then fix it everywhere in separate patch.